code/game/gamemodes/nuclear/nuclearbomb.dm

```diff
@@ -23,18 +23,17 @@ var/obj/item/weapon/disk/nuclear/nukedisk
 
 /obj/machinery/nuclearbomb/New()
 	..()
-	r_code = "[rand(10000, 99999.0)]"//Creates a random code upon object spawn.
+	r_code = "[rand(10000, 99999)]"//Creates a random code upon object spawn.
 
 /obj/machinery/nuclearbomb/process()
-	if (src.timing)
+	if(timing)
 		bomb_set = 1 //So long as there is one nuke timing, it means one nuke is armed.
-		src.timeleft--
-		if (src.timeleft <= 0)
+		timeleft -= SS_WAIT_MACHINERY/10 //Machinery does NOT tick every 10 ms. Divided by ten to convert into seconds
+		if(timeleft <= 0)
 			explode()
 		for(var/mob/M in viewers(1, src))
-			if ((M.client && M.machine == src))
-				src.attack_hand(M)
-	return
+			if((M.client && M.machine == src))
+				attack_hand(M)
```

Comment:
I know this isn't your code but golly gee this is bad

Reply:
I know right ? Shit's raw... which is probably why that dumb bug stayed in here for years